include/msgpack/adaptor/boost/msgpack_variant.hpp

```diff
@@ -0,0 +1,391 @@
+//
+// MessagePack for C++ static resolution routine
+//
+// Copyright (C) 2015 KONDO Takatoshi
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+//
+#ifndef MSGPACK_TYPE_BOOST_MSGPACK_VARIANT_HPP
+#define MSGPACK_TYPE_BOOST_MSGPACK_VARIANT_HPP
+
+#if defined(MSGPACK_USE_BOOST)
+
+#include "msgpack/versioning.hpp"
+#include "msgpack/adaptor/adaptor_base.hpp"
+#include "msgpack/adaptor/check_container_size.hpp"
+#include "msgpack/adaptor/boost/string_ref.hpp"
+
+#include "msgpack/adaptor/nil.hpp"
+#include "msgpack/adaptor/bool.hpp"
+#include "msgpack/adaptor/int.hpp"
+#include "msgpack/adaptor/float.hpp"
+#include "msgpack/adaptor/string.hpp"
+#include "msgpack/adaptor/vector_char.hpp"
+#include "msgpack/adaptor/raw.hpp"
+#include "msgpack/adaptor/ext.hpp"
+#include "msgpack/adaptor/vector.hpp"
+#include "msgpack/adaptor/map.hpp"
+
+#include <boost/variant.hpp>
+
+namespace msgpack {
+
+/// @cond
+MSGPACK_API_VERSION_NAMESPACE(v1) {
+/// @endcond
+
+namespace type {
+
+typedef boost::make_recursive_variant<
+    nil,               // NIL
+    bool,              // BOOL
+    int64_t,           // NEGATIVE_INTEGER
```

Comment:
True. The following rule is applied.

### Packing

|msgpack::type::variant|actual value|msgpack bytes|
|----|---|--------|-----------|
|uint64_t|positive integer|positive fixnum or uint N|
|int64_t|positive integer|positive fixnum or uint N|
|int64_t|negative integer|negative fixnum or int N|

### Unpacking

|msgpack bytes|actual value|msgpack::object::type|
|----|---|--------|
|positive fixnum|positive integer|POSITIVE_INTEGER|
|negative fixnum|negative integer|NEGATIVE_INTEGER|
|int N|positive integer|POSITIVE_INTEGER|
|int N|negative integer|NEGATIVE_INTEGER|
|uint N|positive integer|POSITIVE_INTEGER|

### Converting

|msgpack::object::type|actual value|msgpack::type::variant|
|----|---|--------|
|POSITIVE_INTEGER|positive integer|uint64_t|
|NEGATIVE_INTEGER|positive integer|int64_t|

### Creating object::with_zone

|msgpack::type::variant|actual value|msgpack::object::type|
|----|---|--------|
|int64_t|positive integer|POSITIVE_INTEGER|
|int64_t|negative integer|NEGATIVE_INTEGER|
|uint64_t|positive integer|POSITIVE_INTEGER|

So the comment should be replaced
```C++
    int64_t,           // NEGATIVE_INTEGER
```
with something like
```C++
    int64_t,           // NEGATIVE_INTEGER if actual value < 0 otherwise POSITIVE_INTEGER
```